common/src/main/java/com/wynntils/overlays/StatusEffectsOverlay.java

```diff
@@ -54,8 +64,23 @@ public void onStatusChange(StatusEffectsChangedEvent event) {
     }
 
     private void recalculateRenderCache() {
-        renderCache = Models.StatusEffect.getStatusEffects().stream()
-                .map(statusTimer -> new TextRenderTask(statusTimer.asString(), getTextRenderSetting()))
+        List<StatusEffect> effects = Models.StatusEffect.getStatusEffects();
+        Stream<StatusEffectWithProperties> effectWithProperties;
+
+        if (doStackEffects.get()) {
+            effectWithProperties = stackEffects(effects);
+        } else {
+            effectWithProperties = effects.stream().map(StatusEffectWithProperties::new);
+        }
+
+        if (doSortEffects.get()) {
+            // Sort effects based on their prefix and their name
+            effectWithProperties = effectWithProperties.sorted(Comparator.comparing(
+                    e -> e.getPrefix().getString() + e.getName().getString()));
```

Comment:
Instead of doing this in-line, please make `StatusEffect` implement `Comparator<StatusEffect>`. I advise you to use `ComparisonChain` in your implementation. Also, perhaps we could also sort by modifier?

Reply:
Sorting by modifier is a nice idea. I'd think the sorting priority could be as follows:

1. prefix (the icon)
2. name (to group similar effects together)
3. modifier

Reply:
I have applied most of the changes, though I am not familiar with the `Comparator` interface. So I am assuming my current usage is incorrect.

I implemented the interface like this:

```java
        @Override
        public int compare(StatusEffect effect, StatusEffect t1) {
            return ComparisonChain.start()
                    .compare(effect.getPrefix().getString(), t1.getPrefix().getString())
                    .compare(effect.getName().getString(), t1.getName().getString())
                    .compare(effect.getModifier().getString(), t1.getModifier().getString())
                    .result();
}
```

Which required me to use it like this:

```java
            effectWithProperties = effectWithProperties.sorted(((t1, t2) -> t1.compare(t1.getEffect(), t2.getEffect())));
```

So I'm guessing I don't really understand it

Reply:
Sorry for misleading you. The correct interface is `Comparable`, which will allow you to use `.sorted(Comparator.comparing(...))`.

The chain implementation itself looks good, although I am not sure how well string-sorting works for the modifier, which is a number by essence.

Reply:
We can extract the first number in the modifier like this:
```java
        public static Double getModifierNumber(StyledText modifier){
            Matcher m = DOUBLE_PATTERN.matcher(modifier.getStringWithoutFormatting());
            if( !m.find() ){
                return 0.0;
            }
            String numberStr = DOUBLE_PATTERN.matcher(modifier.getStringWithoutFormatting()).group();
            return Double.parseDouble(numberStr);
        }
```
and use it like this:

```java
        @Override
        public int compareTo(@NotNull StatusEffect effect) {
            return ComparisonChain.start()
                    .compare(this.effect.getPrefix().getString(), effect.getPrefix().getString())
                    .compare(this.effect.getName().getString(), effect.getName().getString())
                    .compare(getModifierNumber(this.effect.getModifier()), getModifierNumber(effect.getModifier()))
                    .result();
        }
```

Which would use the first number in the modifier string and compare it

Reply:
No, we shouldn't do that. It's messy, adds a new regex, etc. Just sort the string for now, and if it becomes problematic, we will see and fix it.